packages/govern-console/src/containers/NewExecution/ActionList.tsx

```diff
@@ -0,0 +1,151 @@
+import React, { memo } from 'react';
+import {
+  TextInput,
+  Card,
+  Box,
+  useTheme,
+  useLayout,
+  ButtonIcon,
+  IconUp,
+  IconDown,
+  IconCross,
+  GU,
+} from '@aragon/ui';
+import { ActionItem } from 'utils/types';
+import { Controller, useFormContext } from 'react-hook-form';
+import { getTruncatedAccountAddress } from 'utils/account';
+
+type ActionListProps = {
+  actions: Array<ActionItem & { id: string }>;
+  swap: (indexA: number, indexB: number) => void;
+  remove: (index: number) => void;
+};
+
+type ActionHeaderProps = {
+  contractAddress: string;
+  index: number;
+  count: number;
+  swap: (indexA: number, indexB: number) => void;
+  remove: (index: number) => void;
+};
+
+const ActionHeader: React.FC<ActionHeaderProps> = memo(function ActionHeader({
+  contractAddress,
+  index,
+  count,
+  swap,
+  remove,
+}) {
+  const theme = useTheme();
+  const { layoutName } = useLayout();
+
+  return (
+    <div
+      style={{
+        display: 'grid',
+        justifyContent: 'space-between',
+        alignItems: 'center',
+        width: '100%',
+        gridAutoFlow: `${layoutName === 'small' ? 'row' : 'column'}`,
+        gridTemplateColumns: `${layoutName === 'small' ? '1fr' : 'auto 100px'}`,
+      }}
+    >
+      <div>
+        Contract:{' '}
+        <span>
+          {layoutName === 'small' ? getTruncatedAccountAddress(contractAddress) : contractAddress}
+        </span>
+      </div>
+      <div
+        style={{
+          display: 'grid',
+          gridAutoFlow: 'column',
+          justifyContent: `${layoutName === 'small' ? 'space-between' : 'end'}`,
+        }}
+      >
+        <div>
+          <ButtonIcon label="Up" onClick={index > 0 ? () => swap(index, index - 1) : undefined}>
+            <IconUp color={theme.primary} />
+          </ButtonIcon>
+          <ButtonIcon
+            label="Down"
+            onClick={index < count - 1 ? () => swap(index, index + 1) : undefined}
+          >
+            <IconDown color={theme.primary} />
+          </ButtonIcon>
+        </div>
+        <div>
+          <ButtonIcon label="Remove" onClick={() => remove(index)}>
+            <IconCross color={theme.red} />
+          </ButtonIcon>
+        </div>
+      </div>
+    </div>
+  );
+});
+
+const ActionList: React.FC<ActionListProps> = ({ actions, swap, remove }) => {
+  const { control } = useFormContext();
+
+  if (actions && actions.length === 0) {
+    return (
+      <Card width="auto" height={`${15 * GU}px`}>
+        No action yet.
+      </Card>
+    );
+  }
+
+  return (
+    <div>
+      {actions.map((action, index: number) => (
+        <Box
+          css={`
+            & > h1 {
+              height: auto !important;
```

Comment:
@Rekard0 this is a hack to overwrite the fixed height heading of a box in the aragon ui.  If you think we should move this to the aragon ui to allow variable length heading, let me know, i'll move it.